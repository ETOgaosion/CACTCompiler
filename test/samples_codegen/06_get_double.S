iobuff:
	.zero 80
.text
.align 1
.globl	main
.type	main, @function
main:
	sd	ra, -8(sp)
	fmv.d.x	ft0, zero
	fsd	ft0, -16(sp)
	addi	sp, sp, -24
	call	get_double
	addi	sp, sp, 24
	fsd	fa0, -24(sp)
	fld	ft0, -24(sp)
	fsd	ft0, -16(sp)
	fld	fa0, -16(sp)
	addi	sp, sp, -24
	call	print_double
	addi	sp, sp, 24
	mv	a0, zero
	ld	ra, -8(sp)
	ret
