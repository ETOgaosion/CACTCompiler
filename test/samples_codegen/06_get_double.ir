============ global var ================
============ imm value ================
============ functions ================
----------- funcDef -----------
main() => 2:
----------- paramVars -----------
----------- localVars -----------
a := symbol type: 1; data type: 4; initValue: 0

----------- tempVars ------------
t_0:= symbol type: TEMP VAR; data type: 4
----------- Codes ------------
main:
	a = 0;
	call get_double; 
	t_0 =  return ;
	t_1 <= a;
	t_1 = t_0;
	addparam t_1;
	call print_double; 
	return 0;

