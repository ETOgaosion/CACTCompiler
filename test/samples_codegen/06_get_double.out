bbl loader
please enter an double number:
5.745000
